tests/unit_tests/testsettingsholder.cpp

```diff
@@ -0,0 +1,120 @@
+        QVERIFY(!settingsHolder.has());                                        \
+        return;                                                                \
+      }                                                                        \
+      case QMetaType::QString: {                                               \
+        QVERIFY(!settingsHolder.has());                                        \
+        QCOMPARE(QVariant(settingsHolder.getter()).toString(), v.toString());  \
+                                                                               \
+        QString expectedValue = "astring";                                     \
+        settingsHolder.setter(QVariant(expectedValue).toType());               \
+        QCOMPARE(spy.count(), 1);                                              \
+        QCOMPARE(QVariant(settingsHolder.getter()).toString(), expectedValue); \
+                                                                               \
+        settingsHolder.remover();                                              \
+        QCOMPARE(spy.count(), 2);                                              \
+        QVERIFY(!settingsHolder.has());                                        \
+        return;                                                                \
+      }                                                                        \
+      case QMetaType::QStringList: {                                           \
+        QVERIFY(!settingsHolder.has());                                        \
+        QCOMPARE(QVariant(settingsHolder.getter()).toStringList().join(","),   \
+                 v.toStringList().join(","));                                  \
+                                                                               \
+        QStringList expectedValue = QStringList({"a", "string", "list"});      \
+        settingsHolder.setter(QVariant(expectedValue).toType());               \
+        QCOMPARE(spy.count(), 1);                                              \
+        QCOMPARE(QVariant(settingsHolder.getter()).toStringList().join(","),   \
+                 expectedValue.join(","));                                     \
+        QVERIFY(settingsHolder.has());                                         \
+                                                                               \
+        settingsHolder.remover();                                              \
+        QCOMPARE(spy.count(), 2);                                              \
+        QVERIFY(!settingsHolder.has());                                        \
+        return;                                                                \
+      }                                                                        \
+      default: {                                                               \
+        Q_ASSERT(false);                                                       \
```

Comment:
Can more information be logged here?

Reply:
Given this is just a test case, do you think a comment would be enough?